src/modules/mavlink/mavlink_mission.h

```diff
@@ -104,16 +104,17 @@ class MavlinkMissionManager
 
 	unsigned		_filesystem_errcount{0};		///< File system error count
 
-	static int8_t		_dataman_id;				///< Global Dataman storage ID for active mission
-	int8_t			_my_dataman_id{0};			///< class Dataman storage ID
+	static dm_item_t		_dataman_id;				///< Global Dataman storage ID for active mission
+	dm_item_t			_my_dataman_id{DM_KEY_WAYPOINTS_OFFBOARD_0};			///< class Dataman storage ID
+
 	static bool		_dataman_init;				///< Dataman initialized
 
 	static uint16_t		_count[3];				///< Count of items in (active) mission for each MAV_MISSION_TYPE
 	static int32_t		_current_seq;				///< Current item sequence in active mission
 
 	int32_t			_last_reached{-1};			///< Last reached waypoint in active mission (-1 means nothing reached)
 
-	int8_t			_transfer_dataman_id{0};		///< Dataman storage ID for current transmission
+	dm_item_t			_transfer_dataman_id{DM_KEY_WAYPOINTS_OFFBOARD_1};		///< Dataman storage ID for current transmission
```

Comment:
This is different - just checking that that logic change is on purpose.

Reply:
Yes it's intentional, but it doesn't really matter either way because it's set before first use to the opposite slot of the active id.